address_book/internal/service/service.go

```diff
@@ -0,0 +1,87 @@
+package service
+
+import (
+	"addressbook/internal/pb"
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type AddressBookService struct {
+	pb.AddressBookServiceServer
+	data sync.Map
+}
+
+func (a *AddressBookService) AddContact(_ context.Context, in *pb.AddContactRequest) (*pb.AddContactResponse, error) {
+	a.data.Store(in.Contact.Phone, in.Contact)
```

Comment:
if in== nil or in.Contact== nil
what will happen in this situation?

Reply:
How can we accept the nill request in grpc ? I'm not sure it's possible